x-pack/plugins/security_solution/cypress/tasks/api_calls/prebuilt_rules.ts

```diff
@@ -109,6 +115,48 @@ export const waitUntilAllRuleAssetsCreated = (
     { interval: 500, timeout: 12000 }
   );
 
+/*
+ * Wait until all rules assets passed in as parameters are
+ * created in the `.kibana_alerting_cases` index as actual installed rules
+ * (saved objects of type alert)
+ */
+export const waitUntilAllRulesCreated = (
+  rules: Array<typeof SAMPLE_PREBUILT_RULE>,
+  index = '.kibana_alerting_cases'
+) =>
+  cy.waitUntil(
+    () => {
+      return cy
+        .request({
+          method: 'GET',
+          url: `${Cypress.env('ELASTICSEARCH_URL')}/${index}/_search`,
+          headers: { 'kbn-xsrf': 'cypress-creds', 'Content-Type': 'application/json' },
+          failOnStatusCode: false,
+          body: {
+            query: {
+              term: {
+                type: {
+                  value: 'alert',
+                },
+              },
+            },
+          },
+        })
+        .then((response) => {
+          const areAllRulesCreated = rules.every((rule) => {
```

Comment:
nit: readability could be improved by moving the logic to a separate function like `doesResponseContainAllRules(response: Cypress.Response<any>, rules: Array<typeof SAMPLE_PREBUILT_RULE>)`. This way `then` section could be simplified to `.then((response) => doesResponseContainAllRules(response, rules)`.